feat(clayui.com): Update content of some components

kresimir-coko opened the pull request:
Addresses some components listed in #3210 

@bryceosterhaus I decided to send this halfway through to make it easier to review and iterate over. I'll be working on the other half of the components listed in the issue next.

### Here's what I've done:
- Add content where it was missing
- Rename `badges` to `badge`
- Rename `text-input-localizable` to `localized-input`
- Update headings to start at h1 (amount of `#` in Markdown)
- Update some navigations where it was needed

### Components included
- Badges
- Breadcrumbs
- Dual List Box
- Loading Indicator
- Localized Input
- Multi Step Nav

### Something I noticed
Localized Input is missing prop to render textarea, this example was included in the Markup/CSS page but we had no support for that. So I removed the example with the idea of eventually adding it back when we add a prop to the component.

bryceosterhaus commented on packages/clay-loading-indicator/README.mdx:
```diff
@@ -15,24 +15,32 @@ import {
 
+The loading indicator shows the user that an external process, like a connection, is being executed.
+
+Use the loading indicator to inform the user that there is pending content from the server that will load where the indicator is displayed.
+Without this indicator, the user can feel lost or navigate away from the page prematurely.
+
```
This info feels redundant from the description.

bryceosterhaus commented on packages/clay-multi-step-nav/README.mdx:
```diff
@@ -14,43 +14,54 @@ import {
 </div>
 
-### ClayMultiStepNav
+# Composing
+
+Multi Step Nav is a progress bar used to divide the main task into subtasks to help users complete long processes.
+Multi Step Nav is used when a major or big task has to be divided into smaller task, with the aim of letting the user breath in the process and providing them with a sense of progression.
```
I wouldn't classify this under the "#composing" header and this info also feels sort of redundant from the "description" in the frontmatter

kresimir-coko replied:
Same as [comment above](https://github.com/liferay/clay/pull/3226#discussion_r425718518). Not rendered in React Component page.

kresimir-coko commented on packages/clay-badge/README.mdx:
```diff
@@ -1,34 +1,39 @@
-<Badge />
-
-We recommend that you review the use cases in the <a href="https://storybook-clayui.netlify.com/?path=/story/claybadge--default">Storybook</a>.
+Badges help highlight important information, such as notifications or new and unread messages.
```
Yes, it's in the description frontmatter of this file, but it's not rendered on the React Component page of Badge
![image](https://user-images.githubusercontent.com/24564752/82042100-9748ee00-96a9-11ea-882d-347ec6a55500.png)

kresimir-coko commented on packages/clay-badge/README.mdx:
```diff
@@ -1,34 +1,39 @@
 <div class="clay-site-alert alert alert-warning">
 	Badges are not used for non-numeric values. If you have a non-numeric value,
-	use labels instead.
+	use labels instead. Badges work for exact numbers up to 999. For numbers
```
It's coming from Lexicon. I wanted to add more content so I just stole some from Lexicon in order to add some more content. Perhaps we need to make it clear what kind of information is contained on these pages.

1. Do we add any advice on how to best use the component?
2. Do we only have technical information on components (props, api)?
3. Do we have any of the content from Lexicon duplicated on Clay, or do we expect users to consult both before using the component?

Keeping in mind that our primary customers are people from DXP, does it hurt having some of the information from Lexicon duplicated on Clay? Sure, some person not attached to Liferay using Clay might get confused by this text, but then again, do we have people outside of Liferay visiting this site.

bryceosterhaus replied:
> Perhaps we need to make it clear what kind of information is contained on these pages.

Yeah this is a great idea. Maybe we could do some research from other sites that that we think have good docs and condense it into some sort of structure for consistency. What would you think about opening a docs ticket where we can add some ideas of what we think would be best to include. These ideas would be similar to your three questions above. I am thinking something along the lines of

1. Overview/Introduction to component. Why/when/where do we use it
2. Variants (going from most used to least used)
3. Limitations/Pitfalls/
3. API Tables


We could do a first pass through and make it as simple as possible with demos, and then fill in areas we see as problematic or need more explaining. By doing this, it would give us a lens to view how we create succinct and consistent docs.

What do you think? If we establish this sort of gird before updating docs, it'll give us a great place to start updating docs.

kresimir-coko commented on packages/clay-loading-indicator/README.mdx:
```diff
@@ -15,24 +15,32 @@ import {
 </div>
 </div>
 
+The loading indicator shows the user that an external process, like a connection, is being executed.
```
Not redundant, these descriptions don't get rendered on React Component pages. Like in the [comment above](https://github.com/liferay/clay/pull/3226#discussion_r425718518).

bryceosterhaus replied:
Ah I see, I think we can remove this check [here](https://github.com/liferay/clay/blob/d5b484fe1889347880b074b795f19593f4501dbf/clayui.com/src/templates/docs.js#L51) and just check for `frontmatter.description` before rendering it. This way we can have descriptions come from frontmatter and have some consistency across files.

kresimir-coko commented:
I've pushed a commit containing the suggestions, also some notes:

- I have discarded the changes to CSS page files as you mentioned
- The descriptions in the frontmatter of MDX files aren't being rendered in React pages, but in CSS pages, which is why I've included them in the content

### Static files keep getting modified
It's been happening for a while now, like a month or even 2, but I keep getting files in `static` folder appear in my git diff when I run any of the `yarn build` scripts (i.e. `yarn develop` in clayui.com folder). [Files in question](https://prnt.sc/shax18), as well as the files in question the [_global-functions.scss](https://github.com/liferay/clay/blob/master/packages/clay-css/src/scss/functions/_global-functions.scss#L1) file gets modified when running `yarn build` in root. 
Is there something I can do about this? Am I missing any configuration, I don't remember changing anything when this started happening.

kresimir-coko commented:
Hey @bryceosterhaus I've made some more changes to this as taken from the [discussion](https://github.com/liferay/clay/issues/3229). Not much to it, I've removed some content that I've added when I thought we needed more of it. Also, I've done the changes to `frontmatter.description` so that they get rendered on both pages as a subheading

pat270 commented:
@kresimir-coko for `_global-functions.scss` I think `yarn format` is modifying the SVG code in the generated `$lx-icons` map during release? It's rounding values `0-32-14.336-32-32s14.299-32` to `0-32-14.336-32-32s14.3-32` (.299-32 to .3-32).

bryceosterhaus commented:
LGTM!
